lhotse/supervision.py

```diff
@@ -97,13 +175,51 @@ def transform_text(self, transform_fn: Callable[[str], str]) -> 'SupervisionSegm
         if self.text is None:
             return self
         return fastcopy(self, text=transform_fn(self.text))
+    
+    def transform_alignment(
+        self,
+        transform_fn: Callable[[str], str],
+        type: Optional[str] = 'word'
+    ) -> 'SupervisionSegment':
+        """
+        Return a copy of the current segment with transformed ``alignment`` field.
+        Useful for text normalization, phonetic transcription, etc.
+
+        :param type:  alignment type to transform (key for alignment dict).
+        :param transform_fn: a function that accepts a string and returns a string.
+        :return: a ``SupervisionSegment`` with adjusted alignments.
+        """
+        if self.alignment is None:
+            return self
+        return fastcopy(
+            self,
+            alignment={
+                ali_type:[
+                    item.transform(transform_fn=transform_fn)
+                    if ali_type == type else item
+                    for item in ali
+                ]
+                for ali_type, ali in self.alignment.items()
+            } if self.alignment else None
```

Comment:
Line 192 already checks whether `self.alignment` is `None`, so no need to do an extra check here.